source/background.ts

```diff
@@ -15,8 +15,11 @@ import {styleHotfixes} from './helpers/hotfix.js';
 
 const {version} = browser.runtime.getManifest();
 
+if(!doesBrowserActionOpenOptions)
+{
 // GHE support
-addPermissionToggle();
+	addPermissionToggle();
```

Comment:
👌

Reply:
Hm, i don't believe `browser.contextMenus` is defined on browsers that don't support them, so maybe a check to see if it is undefined or not could be useful, if there isn't a chrome api solution available?

Reply:
That appears to be accurate: https://github.com/fregante/webext-permission-toggle/pull/26

So you can just drop the mobile/safari check and replace it with `!chrome.contextMenus`

`addPermissionToggle` has its own check inside so this specific change isn't needed here

Reply:
The following error gets thrown on Firefox for android with the `doesBrowserActionOpenOptions` check removed on `addPermissionToggle`


```
00:13:02.267 Uncaught Error: webext-permission-toggle requires the `contextMenus` permission
    addPermissionToggle moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5183
    <anonymous> moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5857
    <anonymous> moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5961
    <anonymous> moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5963
background.js:5183:15
    addPermissionToggle moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5183
    <anonymous> moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5857
    <anonymous> moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5961
    <anonymous> moz-extension://24fb74ed-aa72-4132-a59d-33a31291dced/assets/background.js:5963
```

This prevents the action button from doing anything when pressed.

Reply:
Thanks for checking 🙏 

What does the `permissions` array looks like there? The getter should return the manifest as is, is Firefox altering the manifest and removing the permission that ***does exist*** in the manifest?

Can you try `console.log(chrome.runtime.getManifest())`?

Reply:
According to [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getManifest), the API:

> Returns null for missing values, and removes unsupported keys

This means that it probably removes the `"contextMenus"` permission. This makes it difficult to detect when the author simply forgot to add it. The safest way would be to use `fetch(chrome.runtime.getURL('manifest.json'))` and read it directly, but that's annoying and makes the function async :(

Reply:
- I opened https://github.com/fregante/webext-permission-toggle/issues/52

Reply:
on firefox for android, the contextMenu permission isn't a valid permission, so it is just seems to ignore it if it is in the manifest.

![Screenshot 2024-06-24 005921](https://github.com/refined-github/refined-github/assets/10606102/0f0ffbca-09c9-40f0-a516-aaa816c9c58a)


![Screenshot 2024-06-24 010001](https://github.com/refined-github/refined-github/assets/10606102/cfd77648-4f49-4e24-8647-531d4ddbd7d0)

Reply:
![Screenshot 2024-06-24 010125](https://github.com/refined-github/refined-github/assets/10606102/952ef0da-259e-4f3c-9709-8e2890b861e7)